Fixed #695 Implemented Replicator.remoteUUID API

pasin opened the pull request:
Implemented Replicator.remoteUUID used for representing the remote database instead of URL for the cases (e.g. P2P) that the URLs are not static.

#695

hideki commented on src/main/java/com/couchbase/lite/Manager.java:
```diff
@@ -678,6 +678,11 @@ public Replication getReplicator(Map<String, Object> properties) throws Couchbas
 
+            String remoteUUID = (String) properties.get("remoteUUID");
+            if (remote != null) {
+                repl.setRemoteUUID(remoteUUID);
```
Needs to check if `remoteUUID` is `null`??

pasin replied:
I got wrong. It should check only remoteUUID actually. Will fix this.

hideki commented on src/main/java/com/couchbase/lite/replicator/Replication.java:
```diff
@@ -700,6 +701,31 @@ public void deleteCookie(String name) {
+     */
+    @InterfaceAudience.Public
+    public String getRemoteUUID() {
+        return replicationInternal.getRemoteUUID();
```
`replicationInternal ` could be null?

hideki commented on src/main/java/com/couchbase/lite/replicator/Replication.java:
```diff
@@ -700,6 +701,31 @@ public void deleteCookie(String name) {
+    @InterfaceAudience.Public
+    public void setRemoteUUID(String remoteUUID) {
+        properties.put(ReplicationField.REMOTE_UUID, remoteUUID);
+        replicationInternal.setRemoteUUID(remoteUUID);
```
same here

pasin commented:
@hideki  as the fix per your review is very small, I rebased it to the original commit. I also rebased with the current master. Could you please re-review? Thanks!
